components/datadog/dogstatsd-standalone/k8s.go

```diff
@@ -0,0 +1,258 @@
+	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
+	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
+	v1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/rbac/v1"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+// HostPort defines the port used by the dogstatsd standalone deployment. The
+// standard port for dogstatsd is 8125, but it's already used by the agent.
+const HostPort = 8128
+
+// Socket defines the socket exposed by the dogstatsd standalone deployment.
+// It's not the default to avoid conflict with the agent.
+const Socket = "/var/run/datadog/dsd-standalone.socket"
+
+type K8sComponent struct {
+	pulumi.ResourceState
+}
+
+func K8sAppDefinition(e config.CommonEnvironment, kubeProvider *kubernetes.Provider, namespace string, opts ...pulumi.ResourceOption) (*K8sComponent, error) {
+	opts = append(opts, pulumi.Provider(kubeProvider), pulumi.Parent(kubeProvider), pulumi.DeletedWith(kubeProvider))
+
+	k8sComponent := &K8sComponent{}
+	if err := e.Ctx.RegisterComponentResource("dd:dogstatsd-standalone", "dogstatsd", k8sComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(k8sComponent))
+
+	ns, err := corev1.NewNamespace(
+		e.Ctx,
+		namespace,
+		&corev1.NamespaceArgs{
+			Metadata: metav1.ObjectMetaArgs{
+				Name: pulumi.String(namespace),
+			},
+		},
+		opts...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, utils.PulumiDependsOn(ns))
+
+	daemonSetArgs := appsv1.DaemonSetArgs{
+		Metadata: &metav1.ObjectMetaArgs{
+			Name:      pulumi.String("dogstatsd-standalone"),
+			Namespace: pulumi.String(namespace),
+		},
+		Spec: &appsv1.DaemonSetSpecArgs{
+			Selector: &metav1.LabelSelectorArgs{
+				MatchLabels: pulumi.StringMap{
+					"app": pulumi.String("dogstatsd-standalone"),
+				},
+			},
+			Template: &corev1.PodTemplateSpecArgs{
+				Metadata: &metav1.ObjectMetaArgs{
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("dogstatsd-standalone"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					HostPID:            pulumi.BoolPtr(true),
+					ServiceAccountName: pulumi.String("dogstatsd-standalone"),
+					Containers: corev1.ContainerArray{
+						&corev1.ContainerArgs{
+							Name:  pulumi.String("dogstatsd-standalone"),
+							Image: pulumi.String(e.DogstatsdFullImagePath()),
+							Ports: corev1.ContainerPortArray{
+								&corev1.ContainerPortArgs{
+									ContainerPort: pulumi.Int(8125),
+									HostPort:      pulumi.Int(HostPort),
+									Name:          pulumi.StringPtr("dogstatsdport"),
+									Protocol:      pulumi.StringPtr("UDP"),
+								},
+							},
+							Env: &corev1.EnvVarArray{
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_API_KEY"),
+									Value: e.DogstatsdAPIKey(),
+								},
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_LOG_LEVEL"),
+									Value: pulumi.StringPtr("DEBUG"),
+								},
```

Comment:
For [the agent, we don’t set DEBUG log level](https://github.com/DataDog/test-infra-definitions/blob/8158a5b2c60e9d31f827390168a916a2534c2ac1/components/datadog/agent/helm.go).
So, maybe we can keep the default log level for standalone dogstatsd as well.

Reply:
This is what we have in our k8s-testing-resources repo, but I also think that we should leave the default.
Changed :+1: